internal/pkg/flink/test/mock/gateway_client_fake.go

```diff
@@ -0,0 +1,158 @@
+package mock
+
+import (
+	"fmt"
+	flinkgatewayv1alpha1 "github.com/confluentinc/ccloud-sdk-go-v2/flink-gateway/v1alpha1"
+	"github.com/confluentinc/cli/internal/pkg/ccloudv2"
+	"github.com/confluentinc/cli/internal/pkg/flink/test/generators"
+	"github.com/google/uuid"
+	"pgregory.net/rapid"
+	"time"
+)
+
+const (
```

Comment:
Can you either add some documentation on all fake-gateway features/modes here or to contributing.md?

Or to a local contributing only inside flink package

Reply:
Good idea! I want to avoid having another file in there that needs to be maintained though, so I added some feature descriptions in the comments above the constants 👍.